SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTestRequireHandler.cpp

```diff
@@ -81,267 +81,346 @@ SpatialFunctionalTestRequireHandler::SpatialFunctionalTestRequireHandler()
+		StatusMsg = FString::Printf(TEXT("Received %d %s %d but was expecting A %s B"), A, *OperatorStr, B, *OperatorStr);
+	}
+	else
+	{
+		StatusMsg = FString::Printf(TEXT("Received %d %s %d as expected"), A, *OperatorStr, B);
 	}
 
-	return GenericRequire(Msg, bPassed, ErrorMsg);
+	return GenericRequire(Msg, bPassed, StatusMsg);
 }
 
 bool SpatialFunctionalTestRequireHandler::RequireCompare(float A, EComparisonMethod Operator, float B, const FString& Msg)
 {
 	bool bPassed = Compare(A, Operator, B);
-
-	FString ErrorMsg;
+	FString StatusMsg;
+	FString OperatorStr = GetComparisonMethodAsString(Operator);
 
 	if (!bPassed)
 	{
-		FString OperatorStr = GetComparisonMethodAsString(Operator);
-		ErrorMsg = FString::Printf(TEXT("Received %f %s %f but was expected A %s B"), A, *OperatorStr, B, *OperatorStr);
+		StatusMsg = FString::Printf(TEXT("Received %f %s %f but was expected A %s B"), A, *OperatorStr, B, *OperatorStr);
+	}
+	else
+	{
+		StatusMsg = FString::Printf(TEXT("Received %f %s %f as expected"), A, *OperatorStr, B);
 	}
 
-	return GenericRequire(Msg, bPassed, ErrorMsg);
+	return GenericRequire(Msg, bPassed, StatusMsg);
 }
 
 bool SpatialFunctionalTestRequireHandler::RequireEqual(bool bValue, bool bExpected, const FString& Msg)
 {
 	bool bPassed = bValue == bExpected;
-	FString ErrorMsg;
+	FString StatusMsg;
+	FString ValueStr = bValue ? TEXT("True") : TEXT("False");
 
 	if (!bPassed)
 	{
-		FString ValueStr = bValue ? TEXT("True") : TEXT("False");
 		FString ExpectedStr = bExpected ? TEXT("True") : TEXT("False");
-		ErrorMsg = FString::Printf(TEXT("Received %s but was expecting %s"), *ValueStr, *ExpectedStr);
+		StatusMsg = FString::Printf(TEXT("Received %s but was expecting %s"), *ValueStr, *ExpectedStr);
+	}
+	else
+	{
+		StatusMsg = FString::Printf(TEXT("Received %s as expected"), *ValueStr);
 	}
 
-	return GenericRequire(Msg, bPassed, ErrorMsg);
+	return GenericRequire(Msg, bPassed, StatusMsg);
 }
 
 bool SpatialFunctionalTestRequireHandler::RequireEqual(int Value, int Expected, const FString& Msg)
 {
 	bool bPassed = Value == Expected;
-	FString ErrorMsg;
+	FString StatusMsg;
 
 	if (!bPassed)
 	{
-		ErrorMsg = FString::Printf(TEXT("Received %d but was expecting %d"), Value, Expected);
+		StatusMsg = FString::Printf(TEXT("Received %d but was expecting %d"), Value, Expected);
+	}
+	else
+	{
+		StatusMsg = FString::Printf(TEXT("Received %d as expected"), Value);
 	}
 
-	return GenericRequire(Msg, bPassed, ErrorMsg);
+	return GenericRequire(Msg, bPassed, StatusMsg);
 }
 
 bool SpatialFunctionalTestRequireHandler::RequireEqual(float Value, float Expected, const FString& Msg, float Tolerance)
 {
 	bool bPassed = FMath::Abs(Value - Expected) <= Tolerance;
-	FString ErrorMsg;
+	FString StatusMsg;
 
 	if (!bPassed)
 	{
-		ErrorMsg = FString::Printf(TEXT("Received %f but was expecting %f (tolerance %f)"), Value, Expected, Tolerance);
+		StatusMsg = FString::Printf(TEXT("Received %f but was expecting %f (tolerance %f)"), Value, Expected, Tolerance);
```

Comment:
minor: positive case before negative case?

Reply:
Addressed this in new commit